web/src/settings_org.js

```diff
@@ -1006,6 +1021,35 @@ export function save_discard_widget_status_handler($subsection, for_realm_defaul
     const $save_btn_controls = $subsection.find(".subsection-header .save-button-controls");
     const button_state = show_change_process_button ? "unsaved" : "discarded";
     change_save_button_state($save_btn_controls, button_state);
+
+    // If the stream isn't currently private but being changed to private,
+    // and the user changing this setting isn't subscribed, we show a warning
+    // that they won't be able to access the stream after making it private
+    // unless they subscribe.
+    if (
+        button_state === "unsaved" &&
+        !sub.invite_only &&
+        !sub.subscribed &&
+        switching_to_private(properties_elements, for_realm_default_settings)
+    ) {
+        if ($("#stream_permission_settings .stream_privacy_warning").length > 0) {
+            return;
+        }
+        const context = {
+            banner_type: compose_banner.WARNING,
+            banner_text: $t({
+                defaultMessage:
+                    "You are not currently subscribed to this stream. Unless you subscribe, you will lose access after making it private.",
```

Comment:
The second sentence is a bit awkward; I propose: "You are not subscribed to this stream. Making this stream private without subscribing to it will result in your losing access to it."

or maybe something like:

"Only subscribers can access or join private streams, so you will lose access to this stream if you convert it to a private stream while not subscribed to it."

Reply:
Thanks, changed to the second one!